src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -82,13 +83,21 @@ struct component
 
 static unordered_map<wstring, shared_ptr<component>> g_types;
 
+std::wstring_view wstr_tolower(std::wstring& s) {
```

Comment:
This doesn't seem like the right design (accept a non-const `wstring`, modify it in-place, and return a `wstring_view` to the arg. Either return void, or return the `wstring`.

Reply:
Better yet, since this is only going to be used for the extension, just use it to return a new string containing just the extension (like everything from the '.' to the end).